universal-application-tool-0.0.1/app/assets/javascripts/validation.ts

```diff
@@ -296,15 +308,23 @@ class ValidationController {
     for (const question of nameQuestions) {
       // validate first name is not empty.
       const firstNameInput = <HTMLInputElement>question.querySelector(".cf-name-first input");
-      const firstNameValid = firstNameInput.value.length > 0;
-      this.updateFieldErrorState(question, '.cf-name-first', firstNameValid);
+      const firstNameEmpty = firstNameInput.value.length > 0;
+      this.updateFieldErrorState(question, '.cf-name-first', !firstNameEmpty);
 
       // validate last name is not empty.
       const lastNameInput = <HTMLInputElement>question.querySelector(".cf-name-last input");
-      const lastNameValid = lastNameInput.value.length > 0;
-      this.updateFieldErrorState(question, '.cf-name-last', lastNameValid);
+      const lastNameEmpty = lastNameInput.value.length == 0;
+      this.updateFieldErrorState(question, '.cf-name-last', !lastNameEmpty);
+
+      // validate last name is not empty.
+      const middleNameInput = <HTMLInputElement>question.querySelector(".cf-name-middle input");
+      const middleNameEmpty = middleNameInput.value.length == 0;
+
+      // If this question isn't required then it's also valid if it is empty.
+      const isOptional = !question.classList.contains(ValidationController.REQUIRED_QUESTION_CLASS);
+      const emptyOptional = isOptional && (!firstNameEmpty && !lastNameEmpty && middleNameEmpty);
 
-      isValid = firstNameValid && lastNameValid;
+      isValid = emptyOptional || !(firstNameEmpty || lastNameEmpty);

```

Comment:
```suggestion
      isValid = emptyOptional || (!firstNameEmpty && !lastNameEmpty);

```

Reply:
Suggestion (non-blocking): this is equivalent, but I feel this is easier to reason about, for me at least.